First pass at adding Honeycomb beeline

tfhartmann opened the pull request.

tfhartmann committed:
First pass at adding Honeycomb beeline

hakamadare commented on Dockerfile:
```diff
@@ -19,6 +19,9 @@ ENV CONSUL_PREFIX=healthcheck
 ENV CONSUL_HTTP_ADDR=""
 ENV CONSUL_HTTP_AUTH=""
 ENV EXTRA_ARGS=""
+#ENV HONEYCOMB_API_KEY="ce8f58cf494fa93b7376d053c577a894"
```
is this an actual API key?  perhaps don’t check into GH 😄

tfhartmann replied:
um... no, that couldn't be valid key... nope, nothing to see here. Thanks for point that out, I revoked that key.

hakamadare commented on main.go:
```diff
@@ -7,18 +7,24 @@ import (
 	"github.com/caarlos0/env"
 	"github.com/urfave/negroni"
+
+  beeline "github.com/honeycombio/beeline-go"
```
i think there’s some weird tab/whitespace handling going on here. consider using [go fmt](https://golang.org/cmd/go/#hdr-Gofmt__reformat__package_sources)?

hakamadare commented on main.go:
```diff
@@ -7,18 +7,24 @@ import (
+	Application  string `env:"HEALTHCHECK_APP" envDefault:"healthcheck"`
+	Status       int    `env:"HEALTHCHECK_STATUS" envDefault:"200"`
+	Metrics      string `env:"HEALTHCHECK_METRICS" envDefault:"{}"`
+	HoneyKey     string `env:"HONEYCOMB_API_KEY" envDefault:"ce8f58cf494fa93b7376d053c577a894"`
```
see above re: checking in API key

tfhartmann replied:
I'm an idiot

hakamadare commented on main.go:
```diff
@@ -27,10 +33,19 @@ type healthcheck struct {
-	defcfg := config{Application: "healthcheck", Status: 200, Metrics: "{}"}
+	defcfg := config{Application: "healthcheck", Status: 200, Metrics: "{}" }
+
+	beeline.Init(beeline.Config{
```
maybe consider breaking the config out into its own variable, to make it easier to parse?

```go
beeCfg := beeline.Config{
    WriteKey: cfg .HoneyKey,
    ...
}

beeline.Init(beeCfg)
```

hakamadare replied:
oh, also, you might consider checking that `cfg.HoneyKey` looks like a valid API key, and only `beeline.Init()` if it is?  i looked at the beeline docs and it’s not clear to me how they handle invalid configs

hakamadare reviewed (commented):
LGTM!  some feedback in comments

tfhartmann committed:
Remove refrences to the API Key

hakamadare commented:
sweet!  well done. remember to squash and merge to get rid of the api key 😄

tfhartmann committed:
feat: adding ability to source secrets from vault

tfhartmann committed:
feat: makes the wrapper authenticate to vault

The wrapper will now try to loginto vault with the default vault role of
hc, get a token and if it can, export the token to the ENV, and start a
version of envconsul that will look into vault for the honeycomb api
secret.

If we aren't running on AWS or the VAULT_TOKEN isn't passed in, start a
version that tries to get it's configs from Consul and don't send events
to Honeycomb (becasue don't have a API Token)
